server/src/main/scala/ch.epfl.scala.index.server/PublishProcess.scala

```diff
@@ -0,0 +1,200 @@
+  private def tmpPath(sha1: String) = tmpBase.resolve(s"$sha1.pom")
+
+  private def computeSha1(data: String): String = {
+
+    val md = java.security.MessageDigest.getInstance("SHA-1")
+    md.digest(data.getBytes("UTF-8")).map("%02x".format(_)).mkString
+  }
+
+  /**
+   * verify the sha1 file hash
+   *
+   * @param path the path to the file to verify
+   * @param sha1 the sha1 hash
+   * @return
+   */
+  private def verifySHA1FileHash(path: Path, sha1: String): Boolean = {
+
+    val source = scala.io.Source.fromFile(path.toFile)
+    val content = source.mkString
+    source.close()
+
+    verifyChecksum(content, sha1)
+  }
+  /**
+   * will compute ans sha1 hash from given string and verify that against
+   * given compare hash
+   *
+   * @param toVerify the string to compute
+   * @param sha1 the verify hash
+   * @return
+   */
+  def verifyChecksum(toVerify: String, sha1: String) = {
+
+    computeSha1(toVerify) == sha1
+  }
+
+  def writeFiles(fileName: String, data: String, credentials: GithubCredentials) = Future {
+
+
+    val tmpName = fileName.replace(".sha1", "").replace(".pom", "")
+    //println(s"Receive: $tmpName")
+    val hash = computeSha1(tmpName)
+    val path = tmpPath(hash)
+
+    if (fileName matches """.*\.pom.sha1""") {
+
+      if (verifySHA1FileHash(path, data)) {
+
+        val pomfilePath = pomPath(data)
+//        println("valid sha1 - move file to pom directory")
+
+        if (!Files.exists(pomfilePath)) {
+
+          Files.move(path, pomfilePath)
+        }
+
+        extractPomData(pomfilePath, data, credentials)
+
+      } else {
+
+        println("invalid sha1 - delete file")
+        Files.delete(path)
+      }
+    } else if (fileName matches """.*\.pom""") {
+
+      if (Files.exists(path)) {
+
+        Files.delete(path)
+      }
+//      println("write file to disk")
+      Files.write(path, data.getBytes(StandardCharsets.UTF_8))
+    } else {
+
+//      println("ignored ...")
+    }
+  }
+
+  def extractPomData(path: Path, sha1: String, githubCredentials: GithubCredentials) = {
+
+    val pom = PomsReader.load(path)
+    val githubRepoExtractor = new GithubRepoExtractor
+    val githubRepo = githubRepoExtractor(pom)
+
+    githubRepo.map {repo =>
+
+      new GithubDownload(Some(githubCredentials), system, materializer).run(repo)
+
+      val bintray = BintraySearch(
+        sha1,
+        None,
+        s"${pom.groupId}:${pom.artifactId}",
+        pom.artifactId,
+        "",
+        0,
+        pom.version,
+        pom.groupId,
+        pom.artifactId,
+        new DateTime()
+      )
+
+      val newProject = ProjectConvert(List((pom, List(bintray)))).head
+      val artifact = newProject.artifacts.head
+
+
+      val search = api.projectPage(artifact.reference)
+      search onSuccess {
+        case Some(project) =>
+
+          println("found")
+          val artifacts = if (project.artifacts.exists(a => a.reference == artifact.reference)) {
```

Comment:
@MasseGuillaume Can you think about the update / new release publish scenario, which data have to be updated. Maybe we can fix here also GitHub reference. Maybe I forget to add something.